api/v1alpha1/kubevirtclustertemplate_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// KubevirtClusterTemplateResource describes the data needed to create a KubevirtCluster from a template.
+type KubevirtClusterTemplateResource struct {
+	ObjectMeta clusterv1.ObjectMeta `json:"metadata,omitempty"`
+	Spec KubevirtClusterSpec `json:"spec"`
+}
+
+// KubevirtClusterTemplateSpec defines the desired state of KubevirtClusterTemplate.
+type KubevirtClusterTemplateSpec struct {
+	Template KubevirtClusterTemplateResource `json:"template"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kubevirtclustertemplates,scope=Namespaced,categories=cluster-api,shortName=kct
+
+// KubevirtClusterTemplate is the Schema for the kubevirtclustertemplates API.
+type KubevirtClusterTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KubevirtClusterTemplateSpec `json:"spec,omitempty"`
```

Comment:
could we add something like the validation below to ensure immutability without needing a webhook?
`        // +kubebuilder:validation:XValidation:rule="self == oldSelf", message="spec is immutable"`

Reply:
Hello David, thx for your review :) .
I didn't know it was possible to enforce CRD Immutability with CEL Rules, thanks for the advice. I will update my PR to match your specification.